packages/shared_preferences/shared_preferences_platform_interface/test/shared_preferences_platform_interface_test.dart

```diff
@@ -3,19 +3,33 @@
 // found in the LICENSE file.
 
 import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
 
 void main() {
   TestWidgetsFlutterBinding.ensureInitialized();
 
   group(SharedPreferencesStorePlatform, () {
     test('disallows implementing interface', () {
-      expect(
-        () {
-          SharedPreferencesStorePlatform.instance = IllegalImplementation();
-        },
-        throwsAssertionError,
-      );
+      expect(() {
+        SharedPreferencesStorePlatform.instance = IllegalImplementation();
+      },
+          // In versions of `package:plugin_platform_interface` prior to fixing
+          // https://github.com/flutter/flutter/issues/109339, an attempt to
+          // implement a platform interface using `implements` would sometimes
+          // throw a `NoSuchMethodError` and other times throw an
+          // `AssertionError`. After the issue is fixed, an `AssertionError` will
+          // always be thrown. For the purpose of this test, we don't really care
+          // what exception is thrown, so just allow any exception.
+          throwsA(anything));
```

Comment:
Is this indentation intentional? It looks a little funky

Reply:
This repository uses `dartfmt`. It's an odd side-effect of inserting a comment at this location.

Reply:
Sounds good, resolving